### PR TITLE
Implement ``numeric_only=False`` for ``GroupBy.cumprod`` and ``GroupBy.cumsum``

### PR DESCRIPTION
**Pull request by phofl**
- [ ] Closes #xxxx
- [ ] Tests added / passed
- [ ] Passes `pre-commit run --all-files`

### CODE REVIEW

**Comment by mrocklin** [comment]
I've been seeing a lot of these numeric_only PRs in the last few weeks.  Is
there an issue somewhere explaining what's going on here?  I'm
curious about how important these are.

**Comment by j-bennet** [comment]
@mrocklin 

> I've been seeing a lot of these numeric_only PRs in the last few weeks.  Is
there an issue somewhere explaining what's going on here?  I'm
curious about how important these are.

These are tech debt. When preparing for pandas 2.0, we had to add `numeric_only` kwarg to a lot of `DataFrame` and `DataFrameGroupBy` methods. There were a few large PRs to mimic pandas behavior with warnings on default value, etc. In a lot of methods, actually handling non-numeric data was something we didin't have to do before. Implementing it would take even more time and further inflate the PRs, so we decided to leave that bit for later.

cc @jrbourbeau

File: dask/dataframe/tests/test_groupby.py [hunk 1]
```diff
@@ -3632,3 +3633,48 @@ def test_groupby_numeric_only_true(func):
...
+            "A": 1,
+        }
+    )
+    ddf = dd.from_pandas(df, npartitions=2)
+
+    if PANDAS_GT_200:
+        ctx = pytest.raises(TypeError, match="does not support")
+
+        with ctx:
+            getattr(ddf.groupby("A"), func)(numeric_only=False)
```
Comment (j-bennet) [inline dask/dataframe/tests/test_groupby.py:3654]: Is there a reason to not parametrize with `numeric_only`?

**Review by j-bennet — APPROVED** [review]
👍 , + test `upstream`.

File: dask/dataframe/tests/test_groupby.py (hunk 1 above)
Comment (phofl) [reply dask/dataframe/tests/test_groupby.py:3654]: That would make it a big ugly (imo) in the else branch

**Review by jrbourbeau — APPROVED** [review]

**Pull request merged**